addons/mrp/models/mrp_production.py

```diff
@@ -973,6 +973,10 @@ def _autoconfirm_production(self):
             moves_to_confirm._action_confirm()
             # run scheduler for moves forecasted to not have enough in stock
             moves_to_confirm._trigger_scheduler()
+            if production.picking_type_id.reservation_method == 'at_confirm'\
```

Comment:
Put this in a method _trigger_assign() on mrp.production.

Reply:
with moves_to_assign as parameter

Reply:
would put the method on stock.move in fact.
I would rename the existing method _should_be_assigned to _should_be_assigned_to_picking
and use _should_be_assigned to define all this specific case.